airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -111,4 +125,105 @@ private WbConnectionRead buildWbConnectionRead(ConnectionRead connectionRead) th
     return wbConnectionRead;
   }
 
+  public WbConnectionRead webBackendGetConnection(WebBackendConnectionIdRequestBody webBackendConnectionIdRequestBody)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    final ConnectionIdRequestBody connectionIdRequestBody = new ConnectionIdRequestBody()
+        .connectionId(webBackendConnectionIdRequestBody.getConnectionId());
+
+    final ConnectionRead connection = connectionsHandler.getConnection(connectionIdRequestBody);
+
+    if (webBackendConnectionIdRequestBody.getWithRefreshedCatalog() != null && webBackendConnectionIdRequestBody.getWithRefreshedCatalog()) {
```

Comment:
blarg. been meaning to make this easier. all the json libs seem to like using Boxed Boolean. fore the future! https://github.com/airbytehq/airbyte/pull/1609